packages/s/spdlog/xmake.lua

```diff
@@ -50,7 +50,23 @@ package("spdlog")
             package:add("defines", "SPDLOG_USE_STD_FORMAT")
         elseif package:config("fmt_external") or package:config("fmt_external_ho") then
             package:add("defines", "SPDLOG_FMT_EXTERNAL")
-            package:add("deps", "fmt", {configs = {header_only = package:config("header_only")}})
+            local fmt_version = {
+                ["v1.14.1"] = "10.2.1",
+                ["v1.14.0"] = "10.2.1",
+                ["v1.13.0"] = "10.0.0",
+                ["v1.12.0"] = "10.0.0",
+                ["v1.11.0"] = "9.1.0",
+                ["v1.10.0"] = "8.1.1",
+                ["v1.9.2"] = "8.0.0",
+                ["v1.9.1"] = "8.0.0",
+                ["v1.9.0"] = "8.0.0",
+                default = "7.1.3",
+                ["v1.5.0"] = "6.1.2",
+                ["v1.4.2"] = "6.0.0",
+                ["v1.3.1"] = "5.3.0",
+            }
+            local fmt_ver = fmt_version[tostring(package:version())] or fmt_version.default
```

Comment:
We should not hardcode a specific version number in the package. This can easily lead to various dependency conflict issues.

you can use `add_requireconfs("spdlog.fmt", {version = "7.1.3", override = true})` to modify it's version

Reply:
还有就是 本身spdlog 每个版本雀食要对应 fmt的版本，不然编不过，这个包目前好像没有做这个处理 都是默认dep了最新的fmt

Reply:
> 在另一个package里面 add_deps 也能用这个来 modify么

add_requireconfs 自己随意改。

Reply:
> 还有就是 本身spdlog 每个版本雀食要对应 fmt的版本，不然编不过，这个包目前好像没有做这个处理 都是默认dep了最新的fmt

用户侧 自己走 add_requireconfs 改，要是每个都 deps 里面写死。。依赖一多，各种版本冲突